arkouda/index.py

```diff
@@ -172,16 +173,8 @@ def __ne__(self, other):
         return values != other_values
 
     def _dtype_of_list_values(self, lst):
-        from arkouda.dtypes import dtype
-
         if isinstance(lst, list):
-            d = dtype(type(lst[0]))
-            for item in lst:
-                assert dtype(type(item)) == d, (
-                    f"Values of Index must all be same type.  "
-                    f"Types {d} and {dtype(type(item))} do not match."
-                )
-            return d
+            return np.array(lst).dtype
```

Comment:
Good point.  The only place this is used is to set the dtype of an index that has a list for values.  Since a mixed type list can converted to a pdarray (aparently), I can't think of why this would be a problem?

```
In [7]: l = ["abc", 1, 1.3]

In [8]: np.array(l).dtype
Out[8]: dtype('<U32')

In [9]: ak.array(l)
Out[9]: array(['abc', '1', '1.3'])

```